### DISCUSSION TITLE
PHP 8.1

### DISCUSSION THREAD

**Discussion by luigifab** [OpenMage/magento-lts#1810] [General]
I tried PHP 8.1.0-beta3 (provided by [deb.sury.org](https://deb.sury.org/)).

There is a "bad" change: `Deprecated functionality: trim(): Passing null to parameter #1 ($string) of type string is deprecated  in app/code/core/Mage/Core/Model/Config.php on line 1278`

```php
<?php
echo PHP_VERSION,"\n";
error_reporting(E_ALL);
ini_set('display_errors', 1);

echo strlen(null);
```

**Comment by kiatng** [comment]
https://github.com/OpenMage/magento-lts/blob/4885b20d86bba7aa200007a76620575488e25846/app/code/core/Mage/Core/Model/Config.php#L1274-L1292

An easy fix is change line 1274 to `$className = ''`.

**Reply by luigifab** [reply]
Yes, but there are a lot of errors like this.
I copy past the first.
I'm sad.

**Reply by addison74** [reply]
If we manage to check the changes brought by each new PHP version, I don't think there will be any issues (at least not many). It's true that OpenMage has a lot of shards inside that have accumulated over time but most of it has been solved.

**Comment by addison74** [comment]
I did a search for the error an look what I found: 

https://php.watch/versions/8.1/internal-func-non-nullable-null-deprecation

Most likely in the OpenMage code there are other variables that have assigned null.

**Reply by addison74** [reply]
And here is the list of all information about 8.1

https://php.watch/versions/8.1

**Reply by Flyingmana** [reply]
looks like we will have the same again as with 8.0, there we already hat a set of similar issues related to strong typing

**Comment by kiatng** [comment]
Have anyone checkout https://github.com/Shardj/zf1-future, seems quite active, and the community is updating to PHP8:

> Over 200 updates and bug fixes since 1.12!
The ZF1 community has been hard at work updating Zend Framework with all of the latest features of PHP 7 and 8.0.

If there is a way to port and merge it into OM, it'll make OM lives longer.

**Reply by luigifab** [reply]
I will try it!
